src/main/java/fi/helsinki/cs/tmc/core/communication/oauth2/Oauth.java

```diff
@@ -0,0 +1,49 @@
+package fi.helsinki.cs.tmc.core.communication.oauth2;
+
+public class Oauth {
+
+    private final OauthFlow flow;
+    private String token;
+
+    public Oauth(OauthFlow flow) {
+        this.flow = flow;
+    }
+
+    /**
+     * Returns the oauth token.
+     *
+     * <p>Gets the token from cache or uses the known flow to fetch the it.</p>
+     *
+     * @return oauth token
+     * @throws Exception an error occurred with getting token
+     */
+    public String getToken() throws Exception {
```

Comment:
It seems that probably `throws OAuthSystemException, OAuthProblemException, OAuthRuntimeException` is the longest list needed atm.